### DISCUSSION TITLE
Redirects problem with parenthesis and plus sign

### DISCUSSION THREAD

**Discussion by AnthonyRako** [vercel/next.js#34814] [Help]
Hi everyone,

I am a new Next JS learner. I would like to redirects old links to new links for a site that is already listed on google, using redirects. And in my links I have parenthesis and plus signs
For example: in the function "redirects"
module.exports = {
 async redirects() {
        return [{ source: "/old-blog/abc-+-(default)",
          destination: "/new-blog/abc_+_(default)",
          permanent: true,
        }
       ]
    },
}
I got errors like:
1.`destination` has segments not in `source` or `has` (0) for route {...}, 
2. Error parsing `/old-blog/abc-+-(default)`
                                                       ^ 
Reason: Unexpected MODIFIER at 12, expected END              
source` parse failed for route {…}

I tried to change the parenthesis and plus signs with percent-encoding, and i got no Errors but if i type the url with the parenthesis and plus sign it doesn’t redirect to the desired page. Can anyone please help me? :)

**Comment by wwwdepot** [comment]
You need to escape a few characters, according to the doc https://nextjs.org/docs/api-reference/next.config.js/redirects

**Comment by AnthonyRako** [comment]
I solved it as follows: 
1. in the source : escape "+" , "( "  and  ")"  with "\\" .
2. in the destination: use percent-encoding .

For example: in the function "redirects"
module.exports = {
async redirects() {
return [{ source: "/old-blog/abc-\\+-\\(default\\)",   // this will match  "/old-blog/abc-+-(default)"
destination: "/new-blog/abc_%2B_%28default%29",   // this will match  "/new-blog/abc_+_(default)"
permanent: true,
}
]
},
}
